### DISCUSSION TITLE
Cost tracker for gas

### DISCUSSION THREAD

**Discussion by cpmills1975** [BottlecapDave/HomeAssistant-OctopusEnergy#932] [Q&A]
I've had a current cost display for both electricity and gas on my dashboard since first playing with HA. Originally these were based on readings from my Hildebrand Glow IHD with MQTT combined with a rest sensor for the current rate. This worked perfectly for electricity while I was on tracker and still does for gas (still on tracker for gas).

Having switches to Agile for electricity, I'm now looking at using this integration to calculate up to date costs as this integration knows the rates I am on.

While creating a cost tracker, it seems I am only able to select my electricity meter. Does the cost tracker functionality only work for electricity?  Is is possible to create a cost tracker for gas?  Perhaps not and this is a feature request that needs raising?

**Comment by BottlecapDave** [comment]
Hello. The cost tracker only support electricity meters at the moment, mainly because gas rates only change on a daily basis and has only one rate (e.g. no off peaks), and therefore could easily be calculated with a [template sensor]( https://www.reddit.com/r/homeassistant/comments/uvwd2b/comment/i9nyw4d).

**Comment by cpmills1975** [comment]
Thanks for confirming. I’m using a template sensor at the moment to do exactly that. I find one of the challenges of home assistant however is knowing where a particular sensor is coming from, and having the ability to reduce the number of types of devices and integrations would be helpful.Given the code to calculate the cost over time with multiple rates is already there, would it be very difficult to add the code to support a meter that doesn’t change as often? Surely it is the same code as if I were running tracker or flexible for electricity?Many thanksSent from my iPhoneOn 22 Jun 2024, at 15:20, David Kendall ***@***.***> wrote:﻿
Hello. The cost tracker only support electricity meters at the moment, mainly because gas rates only change on a daily basis and has only one rate (e.g. no off peaks), and therefore could easily be calculated with a template sensor.

**Reply by BottlecapDave** [reply]
Feel free to add a feature request, but because of the reasons stated above, it will be quite low down on my list unless more people request the feature.
